Add Media Resolver

nuuls opened the pull request:
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Add Tooltip-only Resolver for previously unsupported Media files. Particularly useful for file uploaders that do not show the file extension.

![image](https://user-images.githubusercontent.com/17351521/215917859-ccda259e-f000-487a-b0ed-f8202818cd3f.png)

nuuls commented on internal/resolvers/default/link_resolver.go:
```diff
@@ -248,8 +248,10 @@ func New(ctx context.Context, cfg config.APIConfig, pool db.Pool, helixClient *h
-	contentTypeResolvers := []ContentTypeResolver{}
-	contentTypeResolvers = append(contentTypeResolvers, NewPDFResolver(cfg.BaseURL, cfg.MaxContentLength))
+	contentTypeResolvers := []ContentTypeResolver{
+		NewPDFResolver(cfg.BaseURL, cfg.MaxContentLength),
```
Btw @YungLPR mentioned in chat today that this needs to be checked if PDF resolver still works, and it does since it comes first in the array ![1x](https://user-images.githubusercontent.com/17351521/216793110-7f6c071c-262e-42d2-917a-f6f05743e20e.png)

leon-richardt replied:
Yep, noticed that too now :+1: Maybe we can add a comment indicating that more general resolvers should always appear below more specific ones?

pajlada commented:
I would want to think a bit more what data makes sense to pass along to the user.
Size of the file is good, the humanize/bytes class makes sense for this & the tests are nice.
Type of file is good, I would like to know whether the link I'm clicking is going to be an audio file, a video file, or something else.
I don't personally care for the file extension (unless maybe as an extra to the type)
so maybe the tooltip can instead be simplified to two lines:
File type: Video (MP4)
Size: 13.4 MB

File type: Image (PNG)
Size: 2.3 MB

File Type: Text (JSON)?
Size: 0.3 KB

Some tests for the media resolver to ensure we test the various media types we thing are most important would be nice

nuuls commented on internal/resolvers/default/media_resolver.go:
```diff
@@ -0,0 +1,133 @@
+	response := &resolver.Response{
+		Status:  http.StatusOK,
+		Link:    targetURL,
+		Tooltip: tooltip.String(),
```
I removed the PathEscape here and it still works fine in Chatterino. Is there any actual need for it?

nuuls commented:
- Changed the Type and extension into a sinlge line as requested
- Removed handling for `application/*` types as it doesn't really fit the "Media" resolver and is just confusing
- Made file extension handling a bit more reliable and accurate (previous code was stolen from i.nuuls)
- Added tests for the Resolver

![NGRDH](https://user-images.githubusercontent.com/17351521/217076885-6cb42952-575c-489a-bfa6-b8c8b5192354.png)

pajlada commented:
Just need that pathescape back, otherwise works as expected for me
